### PR TITLE
PEP8 compliance

### PR DESCRIPTION
**Pull request by jklenzing**
Went through the current develop branch using a pycodestyle linter to check for PEP8 compliance.  Reduced >3000 non-conforming lines to <150.

### CODE REVIEW

File: pysat/instruments/madrigal_methods.py [hunk 1]
```diff
@@ -212,37 +217,40 @@ def download(date_array, inst_code=None, kindat=None, data_path=None, user=None,
...
+        a = subprocess.check_output(["globalDownload.py", "--verbose",
                                      "--url="+url, '--outputDir='+data_path,
                                      '--user_fullname='+user,
                                      '--user_email='+password,
                                      '--user_affiliation=pysat',
-                                     '--format='+file_format, \
-                            '--startDate='+date_array[0].strftime('%m/%d/%Y'),\
-                            '--endDate='+date_array[-1].strftime('%m/%d/%Y'),
+                                     '--format='+file_format,
+                                     '--startDate=' +
```
Comment (aburrell) [inline pysat/instruments/madrigal_methods.py]: Need spaces about '+' here and elsewhere.

File: pysat/utils.py [hunk 2]
```diff
@@ -367,19 +378,22 @@ def load_netcdf4(fnames=None, strict_meta=False, file_format=None,
                         new_index_name = index_name
                     else:
                         # using integer indexing
-                        new_index = np.arange(loop_lim*step_size, dtype=int) % step_size
+                        new_index = np.arange(loop_lim*step_size,
+                                              dtype=int) % step_size
```
Comment (aburrell) [inline pysat/utils.py]: Need spaces about '*' here and elsewhere

File: demo/cosmic_and_ivm_demo.py [hunk 3]
```diff
@@ -25,41 +26,41 @@ def geo2mag(incoord):
...
+
     """
 
     from numpy import pi, cos, sin, arctan2, sqrt, dot
 
     # SOME 'constants' for location of northern mag pole
-    lat = 80.08#79.3
-    lon = -72.211+360.#287.789 # or 71.41W
+    lat = 80.08  # 79.3
+    lon = -72.211+360.  # 287.789 # or 71.41W
```
[the remark this answers is not in the document]
Comment (jklenzing) [reply demo/cosmic_and_ivm_demo.py]: Fixing.  In general, I've left commented code as is.  The updates here are compliance, not necessarily cleaned code.  Something to discuss at the next meeting?

File: demo/cosmic_and_ivm_demo.py [hunk 4]
```diff
@@ -69,85 +70,94 @@ def geo2mag(incoord):
...
     output = cosmic['edmaxlon'].copy()
     output.name = 'thf2'
-    
-    for i,profile in enumerate(cosmic['profiles']):
-        profile = profile[(profile['ELEC_dens'] >= (1./np.e)*cosmic['edmax'].iloc[i]) & 
-                          (profile.index >= cosmic['edmaxalt'].iloc[i]) ]
+
+    for i, profile in enumerate(cosmic['profiles']):
+        profile = profile[(profile['ELEC_dens'] >=
+                          (1./np.e)*cosmic['edmax'].iloc[i]) &
```
[the remark this answers is not in the document]
Comment (jklenzing) [reply demo/cosmic_and_ivm_demo.py]: fixed

File: demo/cosmic_and_ivm_demo.py [hunk 5]
```diff
@@ -69,85 +70,94 @@ def geo2mag(incoord):
...
+    geo2maglon[1, 0] = -sin(lon)
+    geo2maglon[1, 1] = cos(lon)
+    geo2maglon[2, 2] = 1.
+    out = dot(geo2maglon, xyz)
+
+    tomaglat = np.zeros((3, 3), dtype='float64')
+    tomaglat[0, 0] = cos(.5*pi-lat)
+    tomaglat[0, 2] = -sin(.5*pi-lat)
+    tomaglat[2, 0] = sin(.5*pi-lat)
+    tomaglat[2, 2] = cos(.5*pi-lat)
```
[the remark this answers is not in the document]
Comment (jklenzing) [reply demo/cosmic_and_ivm_demo.py]: fixed

File: pysat/_constellation.py [hunk 6]
```diff
@@ -386,7 +386,9 @@ def difference(self, instrument1, instrument2, bounds, data_labels,
         return { 'data': data_df, 'start':start, 'end':end }
         """
 
-        labels = [dl1 for dl1, dl2 in data_labels] + ['1_'+b[0] for b in bounds] + ['2_'+b[1] for b in bounds] + ['dist']
+        labels = [dl1 for dl1, dl2 in data_labels] + \
+            ['1_'+b[0] for b in bounds] + ['2_'+b[1] for b in bounds] + \
```
[the remark this answers is not in the document]
Comment (jklenzing) [reply pysat/_constellation.py]: fixed

File: pysat/instruments/cosmic2013_gps.py [hunk 7]
```diff
@@ -93,49 +94,56 @@ def list_files(tag=None, sat_id=None, data_path=None, format_str=None):
...
         # preallocate lists
-        year=[None]*num; days=[None]*num; hours=[None]*num; 
-        minutes=[None]*num; microseconds=[None]*num;
-        for i,f in enumerate(cosmicFiles):
+        year = [None]*num
+        days = [None]*num
+        hours = [None]*num
+        minutes = [None]*num
+        microseconds = [None]*num
+        for i, f in enumerate(cosmicFiles):
```
[the remark this answers is not in the document]
Comment (jklenzing) [reply pysat/instruments/cosmic2013_gps.py]: fixed

File: pysat/instruments/cosmic_gps.py [hunk 8]
```diff
@@ -87,50 +87,57 @@ def list_files(tag=None, sat_id=None, data_path=None, format_str=None):
...
         # preallocate lists
-        year=[None]*num; days=[None]*num; hours=[None]*num; 
-        minutes=[None]*num; microseconds=[None]*num;
-        for i,f in enumerate(cosmicFiles):
+        year = [None]*num
+        days = [None]*num
+        hours = [None]*num
+        minutes = [None]*num
+        microseconds = [None]*num
+        for i, f in enumerate(cosmicFiles):
```
[the remark this answers is not in the document]
Comment (jklenzing) [reply pysat/instruments/cosmic_gps.py]: fixed

File: pysat/instruments/madrigal_methods.py (hunk 1 above)
Comment (jklenzing) [reply pysat/instruments/madrigal_methods.py]: fixed

File: pysat/utils.py (hunk 2 above)
Comment (jklenzing) [reply pysat/utils.py]: fixed

**Comment by jklenzing** [comment]
Question:  For an area of code where keywords include operators, which is preferred?  I think that b is technically correct, but it looks weird to me.

a) self._load_data(fid=self._fid-1)
b) self._load_data(fid=self._fid - 1)
c) self._load_data(fid = self._fid - 1)

**Comment by aburrell** [comment]
@jklenzing , I think a and b are technically correct (in that this is not in the examples that I found), but I prefer (a).

**Comment by jklenzing** [comment]
OK.  I've been leaving these as (a).  

@rstoneback, I'm nearly finished with the post-AGB-review commit.  I'm going through the rest of the files to fix the operator issues that @aburrell pointed out.  Do you want me to push this commit before your half of the review, or should I just do one post-review commit?

**Comment by rstoneback** [comment]
Please include your latest commit.

Thanks Jeff.

Cheers,
Russell

On Jan 30, 2019, at 3:04 PM, Jeff Klenzing <notifications@github.com<mailto:notifications@github.com>> wrote:


OK. I've been leaving these as (a).

@rstoneback<https://github.com/rstoneback>, I'm nearly finished with the post-AGB-review commit. I'm going through the rest of the files to fix the operator issues that @aburrell<https://github.com/aburrell> pointed out. Do you want me to push this commit before your half of the review, or should I just do one post-review commit?

**Comment by jklenzing** [comment]
Latest commit pushed.

**Comment by jklenzing** [comment]
I added a unit test from another branch I'm working on to bump the coverage up slightly.  @rstoneback, this is ready for review.

File: pysat/instruments/pysat_sgp4.py [hunk 9]
```diff
@@ -820,37 +949,43 @@ def add_msis(inst, glat_label='glat', glong_label='glong',
...
-    inst.meta['Nn_N2'] = {'units':'cm^-3',
-                          'desc':'Total N2 number particle density from MSIS.'}
-    inst.meta['Nn_O'] = {'units':'cm^-3',
-                         'desc':'Total oxygen number particle density from MSIS.'}
-    inst.meta['Nn_O2'] = {'units':'cm^-3',
-                          'desc':'Total O2 number particle density from MSIS.'}
-    inst.meta['Tn_msis'] = {'units':'K',
-                            'desc':'Neutral temperature from MSIS.'}
+    inst.meta['Nn'] = {'units': 'cm^-3',
+                       'desc': 'Total neutral number particle density ' +
```
Comment (rstoneback) [inline pysat/instruments/pysat_sgp4.py]: Is it ok to use the + operator here? I think in one of my recent reviews of standards I saw that + and strings only works in a particular python interpreter.

File: pysat/instruments/rocsat1_ivm.py [hunk 10]
```diff
@@ -36,33 +36,33 @@
...
 load = cdw.load
 
 # support download routine
 # use the default CDAWeb method
-basic_tag = {'dir':'/pub/data/rocsat/ipei',
-            'remote_fname':'{year:4d}/'+fname,
-            'local_fname':fname}
-supported_tags = {'':{'':basic_tag}}
+basic_tag = {'dir': '/pub/data/rocsat/ipei',
+             'remote_fname': '{year:4d}/'+fname,
```
Comment (rstoneback) [inline pysat/instruments/rocsat1_ivm.py]: I've been using the +, so I am part of, if not the, problem.

File: pysat/instruments/superdarn_grdex.py [hunk 11]
```diff
@@ -164,56 +169,58 @@ def list_files(tag='north', sat_id=None, data_path=None, format_str=None):
 
     if format_str is None and tag is not None:
         if tag == 'north' or tag == 'south':
-            hemi_fmt = ''.join(('{year:4d}{month:02d}{day:02d}.', tag, '.grdex'))
-            return pysat.Files.from_os(data_path=data_path, format_str=hemi_fmt)
+            hemi_fmt = ''.join(('{year:4d}{month:02d}{day:02d}.', tag,
```
Comment (rstoneback) [inline pysat/instruments/superdarn_grdex.py]: This could be the preferred method.

File: pysat/instruments/pysat_sgp4.py (hunk 9 above)
Comment (jklenzing) [reply pysat/instruments/pysat_sgp4.py]: Depending on the code, I've been using a mix.  I'd prefer to use `''.join` everywhere, but that can be tricky.  We can discuss at the next meeting.

**Comment by jklenzing** [comment]
Thanks @rstoneback and @aburrell!  Assuming we are still green after Travis finishes its run, I'll merge this tonight.

File: pysat/instruments/pysat_sgp4.py (hunk 9 above)
Comment (aburrell) [reply pysat/instruments/pysat_sgp4.py]: Another thing, is that we should determine whether it's more pysatic to use 'units' or inst.meta.units_label when setting the meta data

**Comment by rstoneback** [comment]
Units_label is better than ‘units’ since it lets people use a different spelling or even a different word for their metadata, if desired. 

Given the file though, I’m likely the author that didn’t do the best job writing ‘units’. 

Cheers,
Russell

**Pull request merged**